web/src/lib/components/album-page/album-options.svelte

```diff
@@ -1,22 +1,58 @@
 <script lang="ts">
   import Icon from '$lib/components/elements/icon.svelte';
-  import type { AlbumResponseDto, UserResponseDto } from '@immich/sdk';
-  import { mdiClose, mdiPlus } from '@mdi/js';
+  import { updateAlbumInfo, type AlbumResponseDto, type UserResponseDto, AssetOrder } from '@immich/sdk';
+  import { mdiArrowDownThin, mdiArrowUpThin, mdiClose, mdiPlus } from '@mdi/js';
   import { createEventDispatcher } from 'svelte';
 
   import CircleIconButton from '../elements/buttons/circle-icon-button.svelte';
   import FullScreenModal from '$lib/components/shared-components/full-screen-modal.svelte';
   import UserAvatar from '$lib/components/shared-components/user-avatar.svelte';
   import SettingSwitch from '$lib/components/shared-components/settings/setting-switch.svelte';
+  import SettingDropdown from '../shared-components/settings/setting-dropdown.svelte';
+  import type { RenderedOption } from '../elements/dropdown.svelte';
+  import { handleError } from '$lib/utils/handle-error';
 
   export let album: AlbumResponseDto;
   export let user: UserResponseDto;
+  export let onChangeOrder: (order: AssetOrder) => void;
+
+  const options: Record<AssetOrder, RenderedOption> = {
+    [AssetOrder.Asc]: { icon: mdiArrowUpThin, title: 'Ascending' },
+    [AssetOrder.Desc]: { icon: mdiArrowDownThin, title: 'Descending' },
+  };
+
+  $: selectedOption = album.order ? options[album.order] : options[AssetOrder.Desc];
 
   const dispatch = createEventDispatcher<{
     close: void;
     toggleEnableActivity: void;
     showSelectSharedUser: void;
   }>();
+
+  const handleToggle = async (returnedOption: RenderedOption) => {
+    if (selectedOption === returnedOption) {
+      return;
+    }
+    let order = AssetOrder.Desc;
+    for (const [key, option] of Object.entries(options)) {
+      if (option === returnedOption) {
+        order = key as AssetOrder;
+        break;
+      }
+    }
```

Comment:
You can use `findKey(options, (option)  => option === returnedOption)` from lodash-es.